src/MemoryTracing/MemoryInfoProducer.cpp

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "MemoryTracing/MemoryInfoProducer.h"
+
+#include <chrono>
+#include <thread>
+
+#include "MemoryTracingUtils.h"
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/ThreadUtils.h"
+#include "capture.pb.h"
+
+namespace orbit_memory_tracing {
+
+using orbit_grpc_protos::SystemMemoryUsage;
+
+void MemoryInfoProducer::Start() {
+  exit_requested_ = false;
+  thread_ = new std::thread(&MemoryInfoProducer::Run, this);
+}
+
+void MemoryInfoProducer::Stop() {
+  exit_requested_ = true;
+  if (thread_ != nullptr && thread_->joinable()) {
+    thread_->join();
+  }
+  delete thread_;
+  thread_ = nullptr;
+}
+
+void MemoryInfoProducer::Run() {
+  orbit_base::SetCurrentThreadName("MIP::Run");
```

Comment:
Tiny suggestion: "MemInfoPr::Run" so that you fit in 15 characters but still with more info.

Reply:
Thanks for suggesting the thread name! Changed to use "MemInfoPr::Run".